frozen-abi/src/lib.rs

```diff
@@ -19,3 +19,5 @@ extern crate solana_frozen_abi_macro;
 #[cfg(test)]
 #[macro_use]
 extern crate serde_derive;
+
+pub use log;
```

Comment:
i don't think we want this to be public. cc/ @ryoqun

Reply:
I don't _think_ this will work in another crate as macro-laid tokens, then. This might need to get re-worked if this can't be public.

Reply:
I think this can be done with simple in-crate functions, but those would need to be public then, not all of `log`.

Reply:
hmm, seems that `pub`-ing isn't avoidable unfortunately...

how about like this?:

```rust
// Not public API. Referenced by macro-generated code.
#[doc(hidden)]
pub mod __private {
    #[doc(hidden)]
    pub use log;
}
```

refs
- https://internals.rust-lang.org/t/pre-rfc-extend-item-visibility-when-used-by-macro/18967
- https://stackoverflow.com/a/35536199/10242004

Reply:
There's also this, but I'm not sure what's worse, exposing the module or over-string formatting.

```rust

pub mod log {
    pub fn info(msg: &str) {
        log::info!("{}", msg);
    }
    pub fn debug(msg: &str) {
        log::debug!("{}", msg);
    }
    pub fn warn(msg: &str) {
        log::warn!("{}", msg);
    }
    pub fn error(msg: &str) {
        log::error!("{}", msg);
    }
}
```

For now I've committed your suggestion!

Reply:
in this case, we're `pub`-ing under `__private`. so, whole `mod` or subset of its api doesn't matter much to me. thus, fewer loc approach (exposing the module) wins for me